src/Core_Fix_ResourceUnloadOptimizations/Core.ResourceUnloadOptimizations.cs

```diff
@@ -131,6 +173,52 @@ public static AsyncOperation UnloadUnusedAssetsHook()
                 else
                     return RunUnloadAssets();
             }
+
+#if !EC
+            [HarmonyPrefix]
+            [HarmonyPatch(typeof(global::Studio.Studio), nameof(global::Studio.Studio.LoadScene))]
+            [HarmonyPatch(typeof(global::Studio.Studio), nameof(global::Studio.Studio.ImportScene))]
+            [HarmonyPatch(typeof(global::Studio.Studio), nameof(global::Studio.Studio.InitScene))]
+#if !HS && !PH
+            [HarmonyPatch(typeof(global::Studio.Studio), nameof(global::Studio.Studio.LoadSceneCoroutine))]
+#endif
+            public static void LoadScenePrefix()
+            {
+                _sceneLoadOperationsInProgress++;
```

Comment:
What happens if scene loading crashes?

Reply:
Huh... Good question.

Reply:
Updated to reset count when memory pressure check fails. Since outstanding checks will already handle <= 0, this should be fine. I suppose it could cause the odd extra `isLoading` condition if you manage to load another scene within 2-3 frames of the last one